lib/spack/spack/cmd/list.py

```diff
@@ -114,6 +116,11 @@ def match(p, f):
 
         pkgs = [p for p in pkgs if any(match(p, f) for f in res)]
 
+    # If tags have been specified on the command line, filter by tags
+    if args.tags:
+        packages_with_tags = spack.repo.path.packages_with_tags(*args.tags)
+        pkgs = [p for p in pkgs if p in packages_with_tags]
+
```

Comment:
The name of this function is `filter_by_name`, which makes adding the additional filter by tags seem out of place.  I suggest moving this code to the `list` method *before* the `if args.update:` statement.